pyemd/emd.pyx

```diff
@@ -117,37 +119,45 @@ def array_emd(first_array,
             the resulting distance is not guaranteed to be a metric). The default
             value is -1, which means the maximum value in the distance matrix is
             used.
-        bins (int): The number of bins to include in the generated histogram.
-            The default value is 30.
-        distance (string or function): A string or function implementing a 
-            metric on a 1D np.ndarray. Default to euclidean distance. 
+        bins (int or string): The number of bins to include in the generated
+            histogram. If a string, must be one of the bin selection algorithms
+            accepted by `np.histogram`. Defaults to 'auto', which gives the
+            maximum of the ‘sturges’ and ‘fd’ estimators.
+        distance (string or function): A string or function implementing a
+            metric on a 1D np.ndarray. Default to euclidean distance.
             Currently limited to 'euclidean' or your own function which takes
             for input a 1D array and returns a square 2D array of pairwise
             distances.
         normalized (boolean): If true, treat histograms as fractions of the
             dataset. If false, treat histograms as counts. In the latter case
             the EMD will vary greatly by array length.
-        range (int min, int max): A tuple of minimum and maximum values for 
-            histogram. Defaults to the range of the union of `first_array` 
+        range (int min, int max): A tuple of minimum and maximum values for
+            histogram. Defaults to the range of the union of `first_array`
             and `second_array`. Note: if the given range is not a superset
             of the default range, no warning will be given.
-            
+
     Returns:
         float: The EMD value.
     """
     if range is None:
-        range = (min(np.min(first_array), np.min(second_array)), 
+        range = (min(np.min(first_array), np.min(second_array)),
                  max(np.max(first_array), np.max(second_array)))
 
+    if type(bins) == str:
```

Comment:
`isinstance(bins, str)` should be preferred over explicitly checking `type` equality.